services/frontend-service/src/ui/components/ProductVersion/ReleaseTrainDialog.tsx

```diff
@@ -0,0 +1,88 @@
+/*This file is part of kuberpult.
+
+Kuberpult is free software: you can redistribute it and/or modify
+it under the terms of the Expat(MIT) License as published by
+the Free Software Foundation.
+
+Kuberpult is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+MIT License for more details.
+
+You should have received a copy of the MIT License
+along with kuberpult. If not, see <https://directory.fsf.org/wiki/License:Expat>.
+
+Copyright 2023 freiheit.com*/
+import * as React from 'react';
+import { useState } from 'react';
+import { Checkbox } from '../dropdown/checkbox';
+import { ConfirmationDialog } from '../dialog/ConfirmationDialog';
+import { useEnvironments } from '../../utils/store';
+
+export type ReleaseTrainDialogProps = {
+    environment: string;
+    onCancel: () => void;
+    open: boolean;
+};
+
+export const ReleaseTrainDialog: React.FC<ReleaseTrainDialogProps> = (props) => {
+    const [selectedEnvs, setSelectedEnvs] = useState<string[]>([]);
+
+    const envsList = useEnvironments();
+    const onConfirm = React.useCallback(() => {
+        setSelectedEnvs([]);
+    }, []);
+
+    const onCancel = React.useCallback(() => {
+        props.onCancel();
+        setSelectedEnvs([]);
+    }, [props]);
+
+    const addEnv = React.useCallback(
+        (env: string) => {
+            const newEnv = env;
+            const indexOf = selectedEnvs.indexOf(newEnv);
+            if (indexOf >= 0) {
+                const copy = selectedEnvs.concat();
+                copy.splice(indexOf, 1);
+                setSelectedEnvs(copy);
+            } else {
+                setSelectedEnvs(selectedEnvs.concat(newEnv));
+            }
+        },
+        [selectedEnvs]
+    );
+
+    return (
+        <ConfirmationDialog
```

Comment:
This seems to be almost the same as `EnvSelectionDialog`.
Can we re-use the EnvSelectionDialog component here?

Reply:
Will adapt it now !